### PR TITLE
Remove adequacy patch lmr [ANT-1933]

### PR DESCRIPTION
**Pull request by payetvin**
TODO: use warnings for removed parameters, only noticed used to avoid CI errors during dev

### CODE REVIEW

File: src/solver/simulation/include/antares/solver/simulation/economy.h [hunk 1]
```diff
@@ -98,8 +98,7 @@ class Economy
     uint pStartTime;
     uint pNbMaxPerformedYearsInParallel;
     std::vector<PROBLEME_HEBDO> pProblemesHebdo;
-    std::vector<std::unique_ptr<Antares::Solver::Optimization::WeeklyOptimization>>
-      weeklyOptProblems_;
+    std::vector<std::unique_ptr<Optimization::WeeklyOptimization>> weeklyOptProblems_;
```
Comment (flomnes) [inline src/solver/simulation/include/antares/solver/simulation/economy.h]: This `unique_ptr` is no longer necessary.
```suggestion
    std::vector<Optimization::WeeklyOptimization> weeklyOptProblems_;
```

**Review by flomnes — COMMENTED** [review]
1. Remove 
-  `LocalMatching::setToZeroOutsideOutsideLinks`
- `LocalMatching::enabled`

They are no longer used.

2. Please move the remaining property directly to class `AdequacyPatchParams`, and remove sub-class LocalMatching

3. Remove exception `AdqPatchDisabledLMR`, it is now dead code.

**Comment by flomnes** [comment]
Local matching is enabled in [this batch](https://github.com/AntaresSimulatorTeam/Antares_Simulator_Tests_NR/tree/main/adequacy-patch-CSR), please make sure to remove all properties related to LMR from relevant configuration files (generaldata.ini) and publish a new version.

**Comment by payetvin** [comment]
https://github.com/AntaresSimulatorTeam/Antares_Simulator_Tests_NR/pull/31

File: src/solver/simulation/economy.cpp [hunk 2]
```diff
@@ -82,14 +82,15 @@ bool Economy::simulationBegin()
...
-                     resultWriter,
-                     simulationObserver_.get());
+            Optimization::WeeklyOptimization weekOpt(options,
+                                                     &pProblemesHebdo[numSpace],
+                                                     study.parameters.adqPatchParams,
+                                                     numSpace,
+                                                     resultWriter,
+                                                     simulationObserver_.get());
+
+            weeklyOptProblems_.push_back(weekOpt);
```
Comment (flomnes) [inline src/solver/simulation/economy.cpp]:
```suggestion
            weeklyOptProblems_.emplace_back(options, /* other args */);
```
See https://en.cppreference.com/w/cpp/container/vector/emplace_back

**Review by flomnes — DISMISSED** [review]

**Review by flomnes — CHANGES_REQUESTED** [review]
Please remove `AdequacyPatchRuntimeData::AdequacyFirstStep`, and fix function `setBoundsForUnsuppliedEnergy` (possible regression)

File: src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp [hunk 3]
```diff
@@ -134,8 +133,7 @@ void setBoundsForUnsuppliedEnergy(PROBLEME_HEBDO* problemeHebdo,
             }
 
             // adq patch: update ENS <= DENS in 2nd run
```
Comment (flomnes) [inline src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: Please remove l.136 - l. 143

**Review by flomnes — CHANGES_REQUESTED** [review]
Please update the migration guide & changelog (removed options)

**Review by flomnes — APPROVED** [review]

**Review by flomnes — APPROVED** [review]

**Pull request merged**